core/services/vrf/v2/integration_helpers_test.go

```diff
@@ -1683,8 +1683,13 @@ func testMaliciousConsumer(
 	// Register a proving key associated with the VRF job.
 	p, err := vrfkey.PublicKey.Point()
 	require.NoError(t, err)
-	_, err = uni.rootContract.RegisterProvingKey(
-		uni.neil, uni.nallory.From, pair(secp256k1.Coordinates(p)))
+	if vrfVersion == vrfcommon.V2Plus {
+		_, err = uni.rootContract.RegisterProvingKey(
+			uni.neil, nil, pair(secp256k1.Coordinates(p)))
+	} else {
+		_, err = uni.rootContract.RegisterProvingKey(
+			uni.neil, &uni.nallory.From, pair(secp256k1.Coordinates(p)))
+	}
```

Comment:
might be worth pulling this chunk into a separate method considering it's repeated 4 times

Reply:
makes sense 👍